src/kili/services/export/format/voc/__init__.py

```diff
@@ -74,42 +70,42 @@ def _process_asset_for_job(
     asset: Dict,
     images_folder: Path,
     labels_folder: Path,
-) -> Tuple[List, List]:
+) -> None:
     # pylint: disable=too-many-locals, too-many-branches
     """
     Process an asset
     """
     frames = {}
     is_frame = False
-    asset_remote_content = []
 
-    idx = leading_zeros = 0
+    idx = 0
     if "jsonResponse" in asset["latestLabel"]:
         number_of_frames = len(asset["latestLabel"]["jsonResponse"])
-        leading_zeros = len(str(number_of_frames))
         for idx in range(number_of_frames):
             if str(idx) in asset["latestLabel"]["jsonResponse"]:
                 is_frame = True
                 frame_asset = asset["latestLabel"]["jsonResponse"][str(idx)]
                 frames[idx] = {"latestLabel": {"jsonResponse": frame_asset}}
 
+    if is_frame:
+        raise NotImplementedError("Export of annotations on videos is not supported yet.")
+
     if not frames:
         frames[-1] = asset
 
     content_frames = content_repository.get_content_frames_paths(asset)
-    video_filenames = []
     content_frame = content_frames[idx] if content_frames else asset["content"]
 
-    width, height = get_asset_dimensions(content_repository, content_frame, is_frame)
-    for idx, frame in frames.items():
-        if is_frame:
-            filename = f"{asset['externalId']}_{str(idx + 1).zfill(leading_zeros)}"
-            video_filenames.append(filename)
-        else:
-            filename = asset["externalId"]
+    filename = asset["externalId"]
+    img_path = images_folder / Path(asset["content"]).name
+    if not os.path.isfile(content_frame):
+        download_file(content_repository, content_frame, img_path)
```

Comment:
In the assets query, we download assets of a batch just after having query that batch. The reason is that we use signed URLS that have a limit valid time. if you have a dataset of 20 000 assets, that you query all assets and then download them all, there is high chance that the last signed URLS will not be valid anymore!
You should download assets with the assets method and process them

Reply:
And you already downloaded them in the export_project function, in which case would they not be downloaded in order to have to redownload them here ?